jstests/list_movies.test.js

```diff
@@ -0,0 +1,175 @@
+/**
+ * @jest-environment jsdom
+ */
+
+
+const $ = require('jquery');
+global.$ = $;
+
+const module = require('../static/planttracer')
+const list_movies_data = module.list_movies_data
+
+global.Audio = function() {
+  this.play = jest.fn();
+};
+
+
+  describe('list_movies_data', () => {
+    beforeEach(() => {
+      jest.clearAllMocks();
+    });
+  
+    test('should create the correct HTML for published movies', () => {
+      const movies = [
+        {
+          movie_id: 1,
+          user_id: 1,
+          published: 1,
+          title: 'Movie Title',
+          description: 'Description',
+          width: 1280,
+          height: 720,
+          total_bytes: 2000000,
+          fps: 30,
+          total_frames: 3600,
+          date_uploaded: 1627689600,
+          deleted: 0,
+          orig_movie: false,
+        }
+      ];
+  
+      global.user_id = 1;
+      global.admin = true;
+      global.user_demo = false;
+      global.PLAY_LABEL = 'Play';
+      global.PLAY_TRACKED_LABEL = 'Play Tracked';
+      global.UNPUBLISH_BUTTON = 'Unpublish';
+      global.PUBLISH_BUTTON = 'Publish';
+      global.DELETE_BUTTON = 'Delete';
+      global.UNDELETE_BUTTON = 'Undelete';
+      global.TABLE_HEAD = '<tr><th>ID</th><th>Name</th><th>Date</th><th>Title</th><th>Description</th><th>Stats</th><th>Actions</th></tr>';
+      global.user_primary_course_id = 2;
+  
+
+      $.fn.html = jest.fn();
+      list_movies_data(movies);
+      expect($.fn.html).toHaveBeenCalledTimes(4);
+  
+      const expectedPublishedHtml = expect.stringContaining('<table>');
```

Comment:
I don't think this is going to tell us whether the generated HTML contains information that corresponds to the input movie data. All it tests is whether there is a table on the page, not what the contents of the table is.